sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Bindings/CacheAfterWriteStream.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.Bindings
+{
+    /// <summary>
+    /// Class representing a Stream for an object that can be cached locally after being written (and before this Stream is being disposed).
+    /// This class can be used to identify if the object is eligible to be cached.
+    /// This is because the object being written is already present in shared memory.
+    /// It is used by azure-webjobs-sdk to write the object into the cache after the object is written to storage and has all the relevant
+    /// properties (e.g. Blob's eTag) populated by the azure-sdk-for-net storage extension.
+    /// </summary>
+    internal class CacheAfterWriteStream : Stream, ICacheAfterWriteStream
+    {
+        // Metadata of where the object is present in shared memory.
+        // When inserting the object into the cache, this will be used.
+        private readonly SharedMemoryMetadata _sharedMemoryMeta;
+
+        // The cache must not be disposed.
+#pragma warning disable CA2213 // Disposable fields should be disposed
+        private readonly IFunctionDataCache _functionDataCache;
+#pragma warning restore CA2213 // Disposable fields should be disposed
+
+        // Blob representing the object that will be written into the cache.
+        private readonly BlobWithContainer<BlobBaseClient> _blob;
```

Comment:
Are we caching this because we want to call some API? or do we need some information about the blob.
I'm asking because this one might be quite heavy to cache (it has HttpPipeline and HttpClient inside).
Other concern would be that this has credentials inside - is it fine to cache? how long does cache live?
If this is long living and managed identity (oauth) is introduced then this thing will be refreshing token in background - is it fine?